src/encoded/static/components/navigation/components/AccountNav.js

```diff
@@ -0,0 +1,207 @@
+ */
+
+/**
+ * React-Bootstrap Dropdown with User Action menu items.
+ *
+ * @todo Refactor this into a BigDropdown menu.
+ */
+export const AccountNav = React.memo(function AccountNav(props){
+    const { session, updateUserInfo, schemas, ...passProps } = props;
+    const { windowWidth, href } = passProps;
+
+    if (!session) { // Render login button
+        return (
+            <Nav className="navbar-acct">
+                <LoginController {...{ updateUserInfo, auth0Options }}>
+                    <LoginNavItem {...{ schemas, session, href, windowWidth }} key="login-register" className="user-account-item" />
+                </LoginController>
+            </Nav>
+        );
+    }
+
+    const { details: userDetails = {}, user_actions: userActions = [] } = JWT.getUserInfo() || {};
+    const { first_name: acctTitle = "Account", email } = userDetails;
+    const acctIcon = (typeof email === 'string' && email.indexOf('@') > -1 && (
+        object.itemUtil.User.gravatar(email, 30, { 'className' : 'account-icon-image' }, 'mm')
+    )) || <i className="account-icon icon icon-user fas" />;
+    const cls = 'user-account-item is-logged-in is-dropdown' + (acctIcon && acctIcon.type === 'img' ? ' has-image' : '');
+    const navItemTitle = (
+        <React.Fragment>
+            { acctIcon }
+            { acctTitle }
+        </React.Fragment>
+    );
+
+    // `navItemProps` contains: href, windowHeight, windowWidth, isFullscreen, testWarning, mounted, overlaysContainer
+    return (
+        <Nav className="navbar-acct">
+            <BigDropdownNavItem {...passProps} {...{ windowWidth, href }} id="account-menu-item"
+                navItemContent={navItemTitle} className={cls}>
+                <UserActionsMenu {...{ userActions, href, updateUserInfo, userDetails, windowWidth }}/>
+            </BigDropdownNavItem>
+        </Nav>
+    );
+});
+AccountNav.propTypes = {
+    'session'         : PropTypes.bool.isRequired,      /** Passed in by App */
+    'href'            : PropTypes.string.isRequired,    /** Passed in by Redux store */
+    'updateUserInfo'  : PropTypes.func.isRequired,      /** Passed in by App */
+    'mounted'         : PropTypes.bool                  /** Passed in by Navigation */
+};
+
+
+function UserActionsMenu(props){
+    const { userActions, href, updateUserInfo, userDetails, onMenuItemClick, windowWidth, windowHeight } = props;
+    const { first_name: firstName = "Account", last_name: lastName = null } = userDetails;
+    const introTitle = firstName + (lastName ? " " + lastName : "");
+
+    const viewProfileAction = _.findWhere(userActions, { id: "profile" });
+    const viewProfileURL = (viewProfileAction && viewProfileAction.href) || "/me";
+    const renderedActions = userActions.map(function(action, idx){
+        const { id: actionID, title: actionTitle } = action;
+        const targetHref = getActionURL(action, href);
+        const isActive = isActionActive(action, href);
+        let prepend = null;
```

Comment:
too minor fix: replacing if-if-if statement w/ if-elif-elif could be better, since actionID is const and enter to single if block.